### PR TITLE
Family Anniversaries refactoring

### PR DESCRIPTION
**Pull request by harrislineage**
@MichalVasut I may need some help with this... I seem to be unable to get .map() to work here:

```
$(`#Jan`).append(`<span>${AllMonths.Jan[i].day} ${getMonth(AllMonths.Jan[i].date)} ${AllMonths.Jan[i].year}
<a href="https://www.wikitree.com/wiki/${AllMonths.Jan[i].id}">${AllMonths.Jan[i].name}</a> 
${AllMonths.Jan[i].text} ${AllMonths.Jan[i].marriage} [<span class="small">
<a href="https://www.wikitree.com/treewidget/${AllMonths.Jan[i].id}/6">share tree image</a>]</span><br>`);
```

### CODE REVIEW

File: views/calendar/calendar.js [hunk 1]
```diff
@@ -130,8 +124,9 @@ window.calendarView = class calendarView extends View {
                         }
                     });
                 }
+
                 // Then we insert the content into the correct month
-                // This could be simplified...
+                // Mapping does not appear to be working here...
                 for (var i = 0; i < x.length; i++) {
                     try {
                         $(`#Jan`).append(`
```
Comment (MichalVasut) [inline views/calendar/calendar.js:132]: @harrislineage are we talking about this part of code?

```js
$(`#Jan`).append(`
<span>${AllMonths.Jan[i].day} ${getMonth(AllMonths.Jan[i].date)}
${AllMonths.Jan[i].year} <a href="https://www.wikitree.com/wiki/${AllMonths.Jan[i].id}">
${AllMonths.Jan[i].name}</a> ${AllMonths.Jan[i].text} ${
    AllMonths.Jan[i].marriage
} [<span class="small"><a href="https://www.wikitree.com/treewidget/${
    AllMonths.Jan[i].id
}/6">share tree image</a>]
</span><br>
`);

// and further ...
```

If yes, I'm little bit struggling here
- 1st - I'm not able to run it locally - I don't know why, but `watchlist`  returns me empty  array on localhost (not empty in production 🤔 )
- 2nd - what does the array `AllMonths.Jan` / `AllMonths.Feb` / ... contain? Already sorted events per month?

Comment (MichalVasut) [reply views/calendar/calendar.js:132]: If that's the case (events sorted per month) wouldn't this work?

```js
for (monthName of MONTHS) { // loop over all months
    const shortName = monthName.slice(0, 3);
    const events = AllMonths[shortName];

    $(`#${shortName}`).append(
        events // loop over all events in the month
            .map(
                (event) => `
                    <div>
                        ${event.day} ${getMonth(event.date)} ${event.year}
                        <a href="https://www.wikitree.com/wiki/${event.id}">${event.name}</a>
                        ${event.text} ${event.marriage} 
                        [<span class="small">
                            <a href="https://www.wikitree.com/treewidget/${event.id}/6">
                                share tree image
                            </a>
                        </span>]
                    </div>`
            )
            .join("")
    );
}
```

**Comment by harrislineage** [comment]
@MichalVasut thanks for the assistance! Interestingly enough, the mapping revealed an error in previous code where not all events were being published through the original iteration. I am seeing more events now in certain months.

File: views/calendar/calendar.js [hunk 2]
```diff
@@ -126,134 +126,37 @@ window.calendarView = class calendarView extends View {
...
-                            AllMonths.Dec[i].id
-                        }/6">share tree image</a>]
-                        </span><br>
-                    `);
-                    } catch {}
+                for (const monthName of MONTHS) {
+                    // loop over all months
+                    const shortName = monthName.slice(0, 3);
+                    let events = AllMonths[shortName];
+                    $(`#${shortName}`).append(
```
Comment (MichalVasut) [inline views/calendar/calendar.js:133]: Does this actualy work? 😁 - I've just written something out of head, without testing it 😁.

**Comment by MichalVasut** [comment]
> where not all events were being published through the original iteration

this is imho because of the way how you've iterated through the months - there was one main loop (for cyccle) that iterated through all months. the new way iterates each month separatelly (using map function).

File: views/calendar/calendar.js [hunk 3]
```diff
@@ -130,285 +124,62 @@ window.calendarView = class calendarView extends View {
...
-                        month: date.split("-")[1],
-                        day: date.split("-")[2],
-                        marriage: `<a href="https://www.wikitree.com/wiki/${link}">${spouse}</a>`,
-                    });
-                } else if (getMonth(date) == "Dec" && getMonth(date) != undefined) {
-                    AllMonths.Dec.push({
+                for (const monthName of MONTHS) {
+                    const [year, month, day] = date.split("-").map((item) => parseInt(item));
+                    if (month === 0 || monthName != MONTHS[month - 1]) continue;
+                    AllMonths[monthName.slice(0, 3)].push({
```
Comment (MichalVasut) [inline views/calendar/calendar.js:172]: I've just realized that this entire for loop is not necessary...

```suggestion
                    const [year, month, day] = date.split("-").map((item) => parseInt(item));                    
                    if (year == 0 || month === 0 || day === 0) return; // we don't know full date -> we don't want his                    
                    AllMonths[MONTHS[month-1].slice(0, 3)].push({ 
```

File: views/calendar/calendar.js (hunk 2 above)
Comment (harrislineage) [reply views/calendar/calendar.js:133]: I think I had to make a small tweak to it - but it worked!

File: views/calendar/calendar.js [hunk 4]
```diff
@@ -166,10 +166,9 @@ window.calendarView = class calendarView extends View {
             }
 
             function pushEvents(person, date, text, link, spouse) {
-                for (const monthName of MONTHS) {
```
Comment (MichalVasut) [inline views/calendar/calendar.js]: Ouch, there is probably one closing curly braces "}" that closed the for loop - please test it.

**Review by bcaseyrls — COMMENTED** [review]
This version didn't work for me. I got a JS error of an illegal break in push Events. There's an extra curly brace about line 182.

I removed the break and extra brace, and the calendar runs ok for me now. Please double check I haven't broken what you intended here.

File: views/calendar/calendar.js (hunk 4 above)
Comment (MichalVasut) [reply views/calendar/calendar.js]: Thanks @bcaseyrls

**Comment by MichalVasut** [comment]
> I removed the break and extra brace, and the calendar runs ok for me now. Please double check I haven't broken what you intended here.

This should be okay. I've suggested change, but forgot about the curly brace at the end of the fot cycle (noticed about this later - unfortunsttelly I cannot check iy myself, because the watchlist returns me at localhost empty array - same thing returns me in production non-empty array 🤔).
Unfortunately the suggestion was applied without reflecting that notice (and probably without testing).

So thanks again.

**Pull request merged**